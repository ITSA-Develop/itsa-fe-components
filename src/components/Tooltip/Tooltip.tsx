import { Tooltip as AntTooltip, TooltipProps } from 'antd';

export const Tooltip = ({ children, ...rest }: TooltipProps) => {
	return <AntTooltip {...rest}>{children}</AntTooltip>;
};
