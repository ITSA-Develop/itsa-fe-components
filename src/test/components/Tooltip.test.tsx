import { render, screen } from '@testing-library/react';
import { Tooltip } from '../../components/Tooltip/Tooltip';

describe('Tooltip component', () => {
	it('renders tooltip content when open is true', () => {
		const { container } = render(
			<Tooltip title="Texto del tooltip" open>
				<button>Hover me</button>
			</Tooltip>,
		);

		expect(screen.getByText('Texto del tooltip')).toBeInTheDocument();
		expect(container).toMatchSnapshot();
	});
});
