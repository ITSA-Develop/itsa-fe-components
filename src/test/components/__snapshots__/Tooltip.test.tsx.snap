// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Tooltip component > renders tooltip content when open is true 1`] = `
<div>
  <button
    aria-describedby="test-id"
    class="ant-tooltip-open"
  >
    Hover me
  </button>
</div>
`;
